WordPress/Classes/ViewRelated/Me/Me Main/VerifyEmailRow.swift

```diff
@@ -0,0 +1,230 @@
+                        .padding(.horizontal, 16)
+                        .padding(.vertical, 8)
+                        .background(
+                            RoundedRectangle(cornerRadius: 8)
+                                .fill(.white)
+                        )
+                    }
+                    .buttonStyle(.plain)
+                    .disabled(!viewModel.state.isButtonEnabled)
+
+                    Spacer()
+                }
+            }
+        }
+        .padding(16)
+        .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .topLeading)
+    }
+}
+
+// This value is not an actual "timeout" value of the verification link. It's just an arbitrary value to prevent
+// users from sending links repeatedly.
+private let verificationLinkTimeout: TimeInterval = 300
+
+@MainActor
+private class VerifyEmailViewModel: ObservableObject {
+    enum State {
+        case needsVerification
+        case sending
+        case sent(Date)
+        case error(Error)
+
+        var title: String {
+            Strings.title
+        }
+
+        var message: String {
+            let email = try? WPAccount.lookupDefaultWordPressComAccount(in: ContextManager.shared.mainContext)?.email ?? ""
+
+            switch self {
+            case .needsVerification, .sending:
+                if let email, !email.isEmpty {
+                    return String(format: Strings.verifyMessage, email)
+                } else {
+                    return Strings.verifyMessageNoEmail
+                }
+
+            case .sent:
+                if let email, !email.isEmpty {
+                    return String(format: Strings.sentMessage, email)
+                } else {
+                    return Strings.sentMessageNoEmail
+                }
+
+            case .error(let error):
+                return error.localizedDescription
+            }
+        }
+
+        var buttonTitle: String {
+            switch self {
+            case .needsVerification:
+                return Strings.sendButton
+            case .sending:
+                return Strings.sendingButton
+            case .sent:
+                return Strings.sentButton
+            case .error:
+                return Strings.retryButton
+            }
+        }
+
+        var isButtonEnabled: Bool {
+            switch self {
+            case .needsVerification, .error: return true
+            case .sending: return false
+            case .sent(let date):
+                return Date().timeIntervalSince(date) >= verificationLinkTimeout
+            }
+        }
+
+        var showsActivityIndicator: Bool {
+            if case .sending = self {
+                return true
+            }
+            return false
+        }
+    }
+
+    private let userID: NSNumber
+
+    private var lastVerificationSentDate: Date? {
+        get {
+            let key = "LastEmailVerificationSentDate-\(userID)"
+            return UserDefaults.standard.object(forKey: key) as? Date
```

Comment:
Good thinking to put it in `UserDefaults.standard` in case they close/reopen the screen or the whole app.